### DISCUSSION TITLE
Feedback on 1.14.0.rc1 with Ruby 3.2 native gem support

### DISCUSSION THREAD

**Discussion by flavorjones** [sparklemotion/nokogiri#2747] [Releases]
If you have feedback on the v1.14.0.rc1 prerelease, we'd love to hear from you!

https://github.com/sparklemotion/nokogiri/releases/tag/v1.14.0.rc1

---

## 1.14.0.rc1 / 2022-12-29

This is a prerelease.

- Please provide feedback on it at https://github.com/sparklemotion/nokogiri/discussions/2747
- Full details on this release are at https://github.com/sparklemotion/nokogiri/blob/main/CHANGELOG.md

Notable changes:

- Introduces native gem support for Ruby 3.2.
- Ends support for Ruby 2.6 and JRuby 9.3
- Official native gem support for `aarch64-linux`
- Experimental native gem support for `arm-linux`
- Pattern matching API

Please note that Ruby 3.2 changes how symbols are resolved on MacOS, so we're particularly interested in hearing feedback from MacOS users about the native (precompiled) gem packages for Ruby 3.2.

We're waiting for the following to do a final release:

- a final release of rake-compiler-dock that supports Ruby 3.2
- Windows CI environment that runs Ruby 3.2.0, and green tests
- feedback to build confidence that our approach to Darwin symbol resolution in the precompiled extension works for everyone

---

sha256 checksums:

```
9cb5140f400c3599ba9da92a338d2384976179a05e15267d5bda27ce5ab5294a  nokogiri-1.14.0.rc1-aarch64-linux.gem
70946d652626925f0e7d8cb0f03e45b81423496e8f0db30dce7aecccc33336c6  nokogiri-1.14.0.rc1-arm-linux.gem
8562ed2a3765f2020e340d4cbde72c582843191b20114eb24bcc313307483873  nokogiri-1.14.0.rc1-arm64-darwin.gem
6359f13cccc526936913bac3515a3d3d7d1ffde5dffff4eadaaf3cad126fc753  nokogiri-1.14.0.rc1-java.gem
c83050f34b78690bd13398ad91cace3c32957614fd7b6e5ce7098386340cd23d  nokogiri-1.14.0.rc1-x64-mingw-ucrt.gem
4d2edec4b79735b648eca12c5d52b52d03f4ffd300ccacf8fda9169227eb6a4d  nokogiri-1.14.0.rc1-x64-mingw32.gem
55d334f128d86a20497a3fa4e24c67098199dc9428a44b77284cff6c95104c35  nokogiri-1.14.0.rc1-x86-linux.gem
79e1b2f25a26a9a4a87d3af64a47be27ee3cc4ec0d02eef681b64cd8b610e2a6  nokogiri-1.14.0.rc1-x86-mingw32.gem
c359c27e275da0f82d09c25f6efafd116546d2c7c43c91e9393567f00aa3064a  nokogiri-1.14.0.rc1-x86_64-darwin.gem
e15e2c0f844bd113e1b409dfb534b7e2c727b10812734064c05a732b6342f28c  nokogiri-1.14.0.rc1-x86_64-linux.gem
9d527b0b6eeb88f48bc90648df11824f247019e08b95f5a75b6ea02cefcf499a  nokogiri-1.14.0.rc1.gem
```

**Comment by tomstuart** [comment]
I tried Nokogiri 1.14.0.rc1 + Ruby 3.2.0 on Intel and Apple Silicon Macs with a couple of projects which use Nokogiri. Both `x86_64-darwin` and `arm64-darwin` work for me. 🎉

**Reply by flavorjones** [reply]
:pray: Thanks, @tomstuart!

**Comment by tekin** [comment]
Tested two Rails projects with Ruby 3.2 making use of Nokogiri 1.14.0.rc1 on an Arm Mac. Both working  as expected 👍

**Reply by flavorjones** [reply]
Thanks for confirming!

**Comment by kapso** [comment]
Nokogiri 1.14.0.rc1 + Ruby 3.2.0 works on Heroku - x86_64-linux

**Reply by flavorjones** [reply]
Great, thank you for confirming!

**Comment by chaadow** [comment]
Nokogiri 1.14.0.rc1 + Ruby 3.2.0 works on Heroku - x86_64-linux and Apple Silicon M1 macs for me.

**Reply by flavorjones** [reply]
Thank you for confirming!

**Comment by xfalcox** [comment]
Nokogiri 1.14.0.rc1 + Ruby 3.2.0 passes all tests in Discourse running in Linux x64.

**Reply by flavorjones** [reply]
Excellent, thank you for confirming.

**Comment by PikachuEXE** [comment]
Nokogiri 1.14.0.rc1 + Ruby 3.2.0 worker on my M1 Pro (`ruby 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin21]`)

**Reply by flavorjones** [reply]
Great to hear! Thank you.

**Comment by laugmanuel** [comment]
Tested a Rails project using Nokogiri 1.14.0.rc1 + Ruby 3.2.0 (`ruby 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin22]`)  on a MacBook Pro M1 and it passes all tests.
CI using AMD x86_64 and the same versions of Nokogiri and Ruby also work fine.

**Reply by flavorjones** [reply]
Thank you for confirming!

**Comment by aaronstillwell** [comment]
Using 1.14.0.rc1 on a linux machine, in a docker container w/ ruby 3.2.0. No issues encountered so far AFAIK.

**Reply by flavorjones** [reply]
Thank you for the note!

**Comment by flavorjones** [comment]
Update on the three blockers to a final release:

- ✅ We have Windows + Ruby 3.2 CI environments and they're green.
- ✅ There have been no reports of anything broken in the RC.
- 🤨  I'm very close to shipping a minor release of rake-compiler-dock with official support for Ruby 3.2. Should be done today.

So I expect to cut a final release of Nokogiri in the next day or so unless something comes up.

**Reply by tekin** [reply]
Thanks for the update Mike 🙏🏽
